stable/external-dns/templates/secret.yaml

```diff
@@ -25,6 +25,14 @@ data:
   cloudflare_api_key: {{ required "cloudflare.apiKey is required if cloudflare.apiToken is not provided" .Values.cloudflare.apiKey | b64enc | quote }}
   {{- end }}
   {{- end }}
+  {{- if eq .Values.provider "designate" }}
+  {{- if .Values.designate.username }}
+  designate_username: {{ .Values.designate.username | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.designate.password }}
+  designate_password: {{ .Values.designate.password | b64enc | quote }}
+  {{- end }}
+  {{- end }}
```

Comment:
this block also indicates that username and password are optional

Reply:
Username and password are optional. My intent with leaving them optional was to not break the existing workflow of specifying authentication variables in extraEnv.